elastic/synthetics-recorder
# chore: update deps and add `lint-staged`

## Summary

The pre-commit hook aspect of the repo leaves a lot to be desired. I've added `lint-staged` which greatly improved the `lint:fix` functionality in terms of making commits to the repo.

Additionally, I've updated several dependencies that could use a bump, and installed missing peer dependencies to remove warning messages.

## Implementation details

There should be no functional change to the recorder.

`lint-staged` should kick in when attempting to commit changes.

## How to validate this change

Ensure the recorder builds/runs ok. Passing CI should also be a positive indicator that this patch is alright.

Make a lint error such as removing one of the copyright headers from a source file, stage it and attempt to commit. The file should automatically be updated to include the header as part of the commit. Don't push test commits.
